### PR TITLE
Fixes empty hostname returned from HostDiscoveryScript

### PR DESCRIPTION
**Pull request by plliao**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [ ] Did you write any tests to validate this change?  
- [x] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Fixes empty hostname returned from HostDiscoveryScript and make sure hostname is not empty.

Context:
When discovery script return nothing, HostDiscoveryScript will treat the empty result i.e. "" string as the available host. 
It will fail training job as launcher cannot ssh on the host with empty address. Discovery script can return nothing when no workers are available or new workers are not ready yet.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

File: horovod/runner/elastic/discovery.py [hunk 1]
```diff
@@ -169,7 +169,9 @@ def find_available_hosts_and_slots(self):
             if ':' in line:
                 host, slots = line.split(':')
                 host_slots[host] = int(slots)
-            else:
+            # Make sure the host is not empty. The discovery script might
+            # return empty string when all workers are not ready or available.
+            elif host:
```
Comment (ashahab) [inline horovod/runner/elastic/discovery.py:174]: Can you add some tests? Also this would result in an empty map being returned to the caller (update_available_hosts) , which would result in the HostManager treating this as an update in the set of hosts, which is not intended.

**Review by ashahab — CHANGES_REQUESTED** [review]
